openedx/core/djangoapps/course_groups/cohorts.py

```diff
@@ -222,7 +237,7 @@ def get_cohort(user, course_key, assign=True):
 
     user.course_groups.add(cohort)
 
-    return cohort
+    return request_cache.data.setdefault(cache_key, cohort)
```

Comment:
@ormsbee I have updated the function to use the `RequestContext`.

However, as far as I understand this, the `RequestContext` will only cleared at the end of the grading tasks?